### ISSUE TITLE
Bug: Entity Bulk Accessing property not working properly 

### ISSUE THREAD

**Issue by austin92jus** [codeigniter4/CodeIgniter4#9200]
### PHP Version

8.2

### CodeIgniter4 Version

4.5

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

apache

### Database

mysql

### What happened?

while i was upgrade from 4.3 to 4.5 I face the issues on Bulk access entity 
early same code working fine 
but now its not working properly 

```php
class Staff extends Entity {
protected $attributes =['name' => null, 'email' => null];
}
```
Model Class
```php
class StaffModel extends Model {
	protected $table = 'staff_address';
	protected $primaryKey = 'id';
	protected $returnType = \App\Domain\Staff\Staff::class;
	protected $useSoftDeletes = false;
	protected $allowedFields = ['name', 'email',
	];
	protected $useTimestamps = true;
}
```
controller
```php
function save(){
$req = [
['name' => 'test','email' => '[testmail.com](http://testmail.com/)'],
['name' => 'testsd','email' => '[testmail1.com](http://testmail1.com/)']
]
$entiy = new Staff($req);
// throw error
```
```
code
:
500
file
:
"/var/www/gemsAdmin/gems-backend/vendor/codeigniter4/framework/system/Entity/Entity.php"
line
:
411
message
:
"CodeIgniter\\Entity\\Entity::__set(): Argument #1 ($key) must be of type string, int given, called in /var/www/gemsAdmin/gems-backend/vendor/codeigniter4/framework/system/Entity/Entity.php on line 162"
title
:
"TypeError"
trace
:
[{file: "/var/www/gemsAdmin/gems-backend/vendor/codeigniter4/framework/system/Entity/Entity.php",…},…]
type
:
"TypeError"
```

### Steps to Reproduce

Try to pass array of array data to any entity 

### Expected Output

out put comes array of array while I call toRawArray() method in entity

**Comment by neznaika0** [comment]
This is not a bug. Look at the entity's documentation

**Comment by kenjis** [comment]
If the code worked with v4.3, it was a bug.
See https://codeigniter4.github.io/CodeIgniter4/models/entities.html#filling-properties-quickly

I don't understand what you want to do:
```php
$req = [
    ['name' => 'test', 'email' => '[testmail.com](http://testmail.com/)'],
    ['name' => 'testsd', 'email' => '[testmail1.com](http://testmail1.com/)']
]
$entiy = new Staff($req);
```

**Comment by austin92jus** [comment]
> If the code worked with v4.3, it was a bug. See https://codeigniter4.github.io/CodeIgniter4/models/entities.html#filling-properties-quickly
> 
> I don't understand what you want to do:
> 
> ```
> $req = [
>     ['name' => 'test', 'email' => '[testmail.com](http://testmail.com/)'],
>     ['name' => 'testsd', 'email' => '[testmail1.com](http://testmail1.com/)']
> ]
> $entiy = new Staff($req);
> ```
> 
> `Staff` or `$entiy` is an entity. Not two entities.

I need to Create a entity with array of array Data 
if I pass 
$req = [
    ['name' => 'test', 'email' => '[testmail.com](http://testmail.com/)'],
    ['name' => 'testsd', 'email' => '[testmail1.com](http://testmail1.com/)']
]
$entiy = new Staff($req);

its throw error 
if I pass a single associate array-like 
$req =  ['name' => 'test', 'email' => '[testmail.com](http://testmail.com/)'];
it's working

**Comment by kenjis** [comment]
> I need to Create a entity with array of array Data

Do you really need to have an entity with properties that the names are integers?

Your sample code is the same as the following:
```php
$req = [
    0 => ['name' => 'test', 'email' => 'testmail.com'],
    1 => ['name' => 'testsd', 'email' => 'testmail1.com']
]
$entiy = new Staff($req);
```
That is the entity has property `0` and `1`.

**Comment by austin92jus** [comment]
I got it,  4.3 have bug on entity 
now it fixed on 4.5
thank you
